Get rid of warnings during the basic build step

**Is your feature request related to a problem? Please describe.**
Following is the output log for `ninja check-buddy` 
```
In file included from ../include/Interface/buddy/dap/AudioContainer.h:24,
                 from ../tests/Interface/core/AudioContainerTest.cpp:23:
../thirdparty/include/AudioFile.h: In instantiation of 'bool AudioFile<T>::decodeWaveFile(std::vector<unsigned char>&) [with T = float]':
../thirdparty/include/AudioFile.h:419:12:   required from 'bool AudioFile<T>::loadFromMemory(std::vector<unsigned char>&) [with T = float]'
../thirdparty/include/AudioFile.h:409:10:   required from 'bool AudioFile<T>::load(std::string) [with T = float; std::string = std::__cxx11::basic_string<char>]'
../thirdparty/include/AudioFile.h:289:3:   required from 'AudioFile<T>::AudioFile(std::string) [with T = float; std::string = std::__cxx11::basic_string<char>]'
../include/Interface/buddy/dap/AudioContainer.h:37:75:   required from 'dap::Audio<T, N>::Audio(std::string) [with T = float; long unsigned int N = 1; std::string = std::__cxx11::basic_string<char>]'
../tests/Interface/core/AudioContainerTest.cpp:29:76:   required from here
../thirdparty/include/AudioFile.h:545:50: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  545 |           sample = (T) reinterpret_cast<float &>(sampleAsInt);
      |                                                  ^~~~~~~~~~~
../thirdparty/include/AudioFile.h: In instantiation of 'bool AudioFile<T>::decodeAiffFile(std::vector<unsigned char>&) [with T = float]':
../thirdparty/include/AudioFile.h:421:12:   required from 'bool AudioFile<T>::loadFromMemory(std::vector<unsigned char>&) [with T = float]'
../thirdparty/include/AudioFile.h:409:10:   required from 'bool AudioFile<T>::load(std::string) [with T = float; std::string = std::__cxx11::basic_string<char>]'
../thirdparty/include/AudioFile.h:289:3:   required from 'AudioFile<T>::AudioFile(std::string) [with T = float; std::string = std::__cxx11::basic_string<char>]'
../include/Interface/buddy/dap/AudioContainer.h:37:75:   required from 'dap::Audio<T, N>::Audio(std::string) [with T = float; long unsigned int N = 1; std::string = std::__cxx11::basic_string<char>]'
../tests/Interface/core/AudioContainerTest.cpp:29:76:   required from here
../thirdparty/include/AudioFile.h:697:50: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  697 |           sample = (T) reinterpret_cast<float &>(sampleAsInt);
      |                                                  ^~~~~~~~~~~
[60/98] Building CXX object lib/Dialect/Bud/CMakeFiles/obj.BuddyBud.dir/BudDialect.cpp.o
In file included from ../lib/Dialect/Bud/BudDialect.cpp:84:
include/Dialect/Bud/BudOpsAttributes.cpp.inc:31:30: warning: 'mlir::LogicalResult generatedAttributePrinter(mlir::Attribute, mlir::AsmPrinter&)' defined but not used [-Wunused-function]
   31 | static ::mlir::LogicalResult generatedAttributePrinter(::mlir::Attribute def, ::mlir::AsmPrinter &printer) {
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../lib/Dialect/Bud/BudDialect.cpp:84:
include/Dialect/Bud/BudOpsAttributes.cpp.inc:19:36: warning: 'mlir::OptionalParseResult generatedAttributeParser(mlir::AsmParser&, llvm::StringRef*, mlir::Type, mlir::Attribute&)' defined but not used [-Wunused-function]
   19 | static ::mlir::OptionalParseResult generatedAttributeParser(::mlir::AsmParser &parser, ::llvm::StringRef *mnemonic, ::mlir::Type type, ::mlir::Attribute &value) {
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~
[87/98] Building CXX object lib/Conversion/LowerDAP/CMakeFiles/obj.LowerDAPPass.dir/LowerDAPPass.cpp.o
../lib/Conversion/LowerDAP/LowerDAPPass.cpp: In member function 'virtual mlir::LogicalResult {anonymous}::DAPBiquadLowering::matchAndRewrite(buddy::dap::BiquadOp, mlir::PatternRewriter&) const':
../lib/Conversion/LowerDAP/LowerDAPPass.cpp:94:11: warning: variable 'a0' set but not used [-Wunused-but-set-variable]
   94 |     Value a0 = rewriter.create<memref::LoadOp>(loc, kernel, ValueRange{c3});
      |           ^~
../lib/Conversion/LowerDAP/LowerDAPPass.cpp: In lambda function:
../lib/Conversion/LowerDAP/LowerDAPPass.cpp:220:17: warning: variable 'a0' set but not used [-Wunused-but-set-variable]
  220 |           Value a0 = builder.create<memref::LoadOp>(loc, kernel,
      |                 ^~
```

**Describe the solution you'd like**
It would be good if these warnings are eliminated **without** affecting compile time and run-time performance.

--- a/lib/Conversion/LowerDAP/LowerDAPPass.cpp
+++ b/lib/Conversion/LowerDAP/LowerDAPPass.cpp
@@ -84,14 +84,14 @@ public:
     Value c0 = rewriter.create<ConstantIndexOp>(loc, 0);
     Value c1 = rewriter.create<ConstantIndexOp>(loc, 1);
     Value c2 = rewriter.create<ConstantIndexOp>(loc, 2);
-    Value c3 = rewriter.create<ConstantIndexOp>(loc, 3);
+    // Value c3 = rewriter.create<ConstantIndexOp>(loc, 3);
     Value c4 = rewriter.create<ConstantIndexOp>(loc, 4);
     Value c5 = rewriter.create<ConstantIndexOp>(loc, 5);
 
     Value b0 = rewriter.create<memref::LoadOp>(loc, kernel, ValueRange{c0});
     Value b1 = rewriter.create<memref::LoadOp>(loc, kernel, ValueRange{c1});
     Value b2 = rewriter.create<memref::LoadOp>(loc, kernel, ValueRange{c2});
-    Value a0 = rewriter.create<memref::LoadOp>(loc, kernel, ValueRange{c3});
+    // Value a0 = rewriter.create<memref::LoadOp>(loc, kernel, ValueRange{c3});
     Value a1 = rewriter.create<memref::LoadOp>(loc, kernel, ValueRange{c4});
     Value a2 = rewriter.create<memref::LoadOp>(loc, kernel, ValueRange{c5});
 
@@ -192,7 +192,7 @@ public:
     Value c0 = rewriter.create<ConstantIndexOp>(loc, 0);
     Value c1 = rewriter.create<ConstantIndexOp>(loc, 1);
     Value c2 = rewriter.create<ConstantIndexOp>(loc, 2);
-    Value c3 = rewriter.create<ConstantIndexOp>(loc, 3);
+    // Value c3 = rewriter.create<ConstantIndexOp>(loc, 3);
     Value c4 = rewriter.create<ConstantIndexOp>(loc, 4);
     Value c5 = rewriter.create<ConstantIndexOp>(loc, 5);
 
@@ -217,8 +217,8 @@ public:
                                                     ValueRange{ivs[0], c1});
           Value b2 = builder.create<memref::LoadOp>(loc, kernel,
                                                     ValueRange{ivs[0], c2});
-          Value a0 = builder.create<memref::LoadOp>(loc, kernel,
-                                                    ValueRange{ivs[0], c3});
+        //   Value a0 = builder.create<memref::LoadOp>(loc, kernel,
+        //                                             ValueRange{ivs[0], c3});
           Value a1 = builder.create<memref::LoadOp>(loc, kernel,
                                                     ValueRange{ivs[0], c4});
           Value a2 = builder.create<memref::LoadOp>(loc, kernel,

--- a/lib/Conversion/LowerDAP/LowerDAPPass.cpp
+++ b/lib/Conversion/LowerDAP/LowerDAPPass.cpp
@@ -84,14 +84,13 @@ public:
     Value c0 = rewriter.create<ConstantIndexOp>(loc, 0);
     Value c1 = rewriter.create<ConstantIndexOp>(loc, 1);
     Value c2 = rewriter.create<ConstantIndexOp>(loc, 2);
-    // Value c3 = rewriter.create<ConstantIndexOp>(loc, 3);
     Value c4 = rewriter.create<ConstantIndexOp>(loc, 4);
     Value c5 = rewriter.create<ConstantIndexOp>(loc, 5);
 
     Value b0 = rewriter.create<memref::LoadOp>(loc, kernel, ValueRange{c0});
     Value b1 = rewriter.create<memref::LoadOp>(loc, kernel, ValueRange{c1});
     Value b2 = rewriter.create<memref::LoadOp>(loc, kernel, ValueRange{c2});
-    // Value a0 = rewriter.create<memref::LoadOp>(loc, kernel, ValueRange{c3});
+    // Value a0 of kernel is not used
     Value a1 = rewriter.create<memref::LoadOp>(loc, kernel, ValueRange{c4});
     Value a2 = rewriter.create<memref::LoadOp>(loc, kernel, ValueRange{c5});
 
@@ -192,7 +191,6 @@ public:
     Value c0 = rewriter.create<ConstantIndexOp>(loc, 0);
     Value c1 = rewriter.create<ConstantIndexOp>(loc, 1);
     Value c2 = rewriter.create<ConstantIndexOp>(loc, 2);
-    // Value c3 = rewriter.create<ConstantIndexOp>(loc, 3);
     Value c4 = rewriter.create<ConstantIndexOp>(loc, 4);
     Value c5 = rewriter.create<ConstantIndexOp>(loc, 5);
 
@@ -217,8 +215,7 @@ public:
                                                     ValueRange{ivs[0], c1});
           Value b2 = builder.create<memref::LoadOp>(loc, kernel,
                                                     ValueRange{ivs[0], c2});
-        //   Value a0 = builder.create<memref::LoadOp>(loc, kernel,
-        //                                             ValueRange{ivs[0], c3});
+          // Value a0 of kernel is not used
           Value a1 = builder.create<memref::LoadOp>(loc, kernel,
                                                     ValueRange{ivs[0], c4});
           Value a2 = builder.create<memref::LoadOp>(loc, kernel,

--- a/thirdparty/include/AudioFile.h
+++ b/thirdparty/include/AudioFile.h
@@ -541,9 +541,10 @@ bool AudioFile<T>::decodeWaveFile(std::vector<uint8_t> &fileData) {
         int32_t sampleAsInt = fourBytesToInt(fileData, sampleIndex);
         T sample;
 
-        if (audioFormat == WavAudioFormat::IEEEFloat)
-          sample = (T) reinterpret_cast<float &>(sampleAsInt);
-        else // assume PCM
+        if (audioFormat == WavAudioFormat::IEEEFloat) {
+          static_assert(sizeof(sampleAsInt) == sizeof(sample));
+          std::memcpy(&sample, &sampleAsInt, sizeof(sampleAsInt));
+        } else // assume PCM
           sample = (T)sampleAsInt /
                    static_cast<float>(std::numeric_limits<std::int32_t>::max());
 
@@ -575,9 +576,10 @@ bool AudioFile<T>::decodeAiffFile(std::vector<uint8_t> &fileData) {
   // Endianness::BigEndian) + 8;
   std::string format(fileData.begin() + 8, fileData.begin() + 12);
 
-  int audioFormat = format == "AIFF"   ? AIFFAudioFormat::Uncompressed
-                    : format == "AIFC" ? AIFFAudioFormat::Compressed
-                                       : AIFFAudioFormat::Error;
+  int audioFormat = format == "AIFF"
+                        ? AIFFAudioFormat::Uncompressed
+                        : format == "AIFC" ? AIFFAudioFormat::Compressed
+                                           : AIFFAudioFormat::Error;
 
   // -----------------------------------------------------------
   // try and find the start points of key chunks
@@ -693,9 +695,10 @@ bool AudioFile<T>::decodeAiffFile(std::vector<uint8_t> &fileData) {
             fourBytesToInt(fileData, sampleIndex, Endianness::BigEndian);
         T sample;
 
-        if (audioFormat == AIFFAudioFormat::Compressed)
-          sample = (T) reinterpret_cast<float &>(sampleAsInt);
-        else // assume uncompressed
+        if (audioFormat == AIFFAudioFormat::Compressed) {
+          static_assert(sizeof(sampleAsInt) == sizeof(sample));
+          std::memcpy(&sample, &sampleAsInt, sizeof(sampleAsInt));
+        } else // assume uncompressed
           sample = (T)sampleAsInt /
                    static_cast<float>(std::numeric_limits<std::int32_t>::max());
 

